File: Groupe2/TP2/src/phrases.py
def search_phrase_in_file(filename, phrase):
    try:
        # Ouvrir le fichier `filename` en mode lecture
        with open(filename, 'r', encoding='utf-8') as file:
            # Lire toutes les lignes du fichier et les stocker dans `lines`
            lines = file.readlines()

        print("\nRésultats de la recherche :")
        found = False

        # Rechercher la phrase dans chaque ligne
        for i, line in enumerate(lines, start=1):
            count = line.count(phrase)  # Compter le nombre d'occurrences
            if count > 0:  # Si une ou plusieurs occurrences trouvées
                print(f"Ligne {i} => {count} occurrence(s)")
                found = True

        if not found:
            print("Aucune occurrence trouvée.")

    except FileNotFoundError:
        print("Erreur : fichier non trouvé.")


if __name__ == "__main__":
    # Demander le nom du fichier et la phrase à rechercher
    filename = input("Entrez le nom du fichier : ")
    phrase = input("Entrez la phrase que vous souhaitez rechercher : ")

    # Exécuter la recherche
    search_phrase_in_file(filename, phrase)

File: Groupe2/TP2/src/phrases.c
#include <stdio.h>
#include <string.h>

int main() {
    char filename[100], phrase[100], line[256];
    FILE *file;
    int line_number = 0, found = 0;

    // Demander le nom du fichier et la phrase à rechercher
    printf("Entrez le nom du fichier : ");
    scanf("%s", filename);
    printf("Entrez la phrase que vous souhaitez rechercher : ");
    getchar();  // Pour consommer le '\n' après scanf
    fgets(phrase, sizeof(phrase), stdin);
    phrase[strcspn(phrase, "\n")] = '\0';  // Supprimer le retour à la ligne

    // Ouvrir le fichier
    file = fopen(filename, "r");
    if (file == NULL) {
        printf("Erreur lors de l'ouverture du fichier.\n");
        return 1;
    }

    printf("\nRésultats de la recherche :\n");

    // Lire le fichier ligne par ligne et rechercher la phrase
    while (fgets(line, sizeof(line), file)) {
        line_number++;  // incrémenter le numéro de ligne
        int count = 0;
        char *pos = line;

        // Compter le nombre d'occurrences de la phrase dans la ligne
        while ((pos = strstr(pos, phrase)) != NULL) {
            count++;
            pos += strlen(phrase);  // Avancer après la phrase trouvée
        }

        if (count > 0) {
            printf("Ligne %d => %d occurrence(s)\n", line_number, count);
            found = 1;
        }
    }

    // Si aucune occurrence n'a été trouvée
    if (!found) {
        printf("Aucune occurrence trouvée.\n");
    }

    // Fermer le fichier
    fclose(file);
    return 0;
}
